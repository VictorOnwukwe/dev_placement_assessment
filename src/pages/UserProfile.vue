<template>
  <div>
    <div class="transition" style="display:flex;margin-top:44px" v-if="user">
      <img
        :src="user.picture.large"
        style="width:180px;height:180px;min-width:180px;min-height:180px;border-radius:50%;border:6px solid var(--image-border);object-fit:cover"
      />
      <div style="margin-left:28px">
        <h2 style="color:var(--primary);margin-top:16px">
          {{
            `${user.name.title}. ${user.name.first} ${user.name.last}`
          }}&nbsp;<span style="font-weight:300">{{ user.dob.age }}</span>
        </h2>
        <p
          style="color:var(--text-secondary);font-weight:400;letter-spacing: -0.09px;"
          class="location"
        >
          {{
            `${user.location.street.number} ${user.location.street.name}, ${
              user.location.city
            }, ${user.location.state}${
              showCountry ? ", " + user.location.country : ""
            }`
          }}
        </p>
        <button
          class="email"
          style="background-color:#E2E3EC;padding:9px 14px;border-radius:999px;align-items:center;display:block"
        >
          <div style="display:flex;align-items:center">
            <i
              class="far fa-envelope"
              style="font-size:16px;margin-right:8px;color:var(--icon-primary)"
            ></i>
            <span
              style="font-size:14px;color:var(--text-hint);letter-spacing: -0.09px;font-weight:600"
              >{{ user.email }}</span
            >
          </div>
        </button>
        <button
          style="background-color:#F7D9F2;font-size:12px;padding:9px 15px;border-radius:999px;color: var(--text-secondary);font-weight:700;margin-top:16px;display:block"
        >
          JOINED: {{ user.registered.date.slice(0, 10) }}
        </button>
        <p style="color: var(--text-hint);font-size:14px;font-weight:500">
          <i
            class="fas fa-phone-alt"
            style="margin-right:8px;color:var(--icon-primary)"
          ></i>
          <span>{{ user.cell }}</span>
        </p>
        <p style="color: var(--text-hint);font-size:14px;font-weight:500">
          <i
            class="fas fa-mobile-alt"
            style="margin-right:8px;margin-left:4px;color:var(--icon-primary)"
          ></i>
          <span>{{ user.phone }}</span>
        </p>
      </div>
    </div>
  </div>
</template>
<script>
export default {
  data() {
    return {
      user: null,
    };
  },
  computed: {
    showCountry() {
      return this.$store.getters["showCountry"];
    },
  },
  mounted() {
    this.user = this.$store.getters["allUsers"].find(
      (user) => user.email == this.$route.params.email
    );
  },
};
</script>
<style scoped>
.absolute {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  z-index: 0;
}
</style>
