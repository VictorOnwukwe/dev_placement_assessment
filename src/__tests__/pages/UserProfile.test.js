import { mount, createLocalVue } from "@vue/test-utils";
import UserProfile from "../../pages/UserProfile";
import Vuex from "vuex";

describe("User Profile Page", () => {
  let populatedUser = {
    name: { first: "firstname", last: "lastname" },
    picture: { medium: "medium" },
    location: {
      street: { number: "streetnumber", name: "streetname" },
      city: "city",
      state: "state",
      country: "country",
    },
    email: "email",
    phone: "phone",
    cell: "cell",
    dob: { age: "age" },
    registered: { date: "registrationdate" },
  };
  let defaultData;
  const localVue = createLocalVue();
  localVue.use(Vuex);
  let store;

  beforeEach(() => {
    store = new Vuex.Store({
      getters: {
        allUsers: () => [populatedUser],
      },
    });

    defaultData = {
      store,
      mocks: {
        $route: { params: { email: "email" } },
      },
      localVue,
    };
  });

  it("Should not render component without user prop", async () => {
    const wrapper = mount(UserProfile, { ...defaultData });
    expect(wrapper).toBeEmpty;
  });

  it("should render correctly when passed user prop", () => {
    const wrapper = mount(UserProfile, {
      computed: { showCountry: jest.fn() },
      ...defaultData,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render correctly when passed user prop", () => {
    const wrapper = mount(UserProfile, {
      computed: { showCountry: jest.fn() },
      ...defaultData,
      data() {
        return {
          user: populatedUser,
        };
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should show country when state value is true", async () => {
    store = new Vuex.Store({
      getters: {
        allUsers: () => [populatedUser],
        showCountry: () => true,
      },
    });

    const wrapper = mount(UserProfile, {
      ...defaultData,
      store,
      data() {
        return {
          user: populatedUser,
        };
      },
    });

    let elem = wrapper.find("p.location");
    expect(elem.text()).toContain("country");
  });

  it("should not show country when state value is false", async () => {
    store = new Vuex.Store({
      getters: {
        allUsers: () => [populatedUser],
        showCountry: () => false,
      },
    });

    const wrapper = mount(UserProfile, {
      ...defaultData,
      store,
      data() {
        return {
          user: populatedUser,
        };
      },
    });

    let elem = wrapper.find("p.location");
    expect(elem.text().includes("country")).toBe(false);
  });
});
