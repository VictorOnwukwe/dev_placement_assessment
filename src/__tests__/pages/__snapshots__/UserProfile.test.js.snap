// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Profile Page should render correctly when passed user prop 1`] = `
<div>
  <!---->
</div>
`;

exports[`User Profile Page should render correctly when passed user prop 2`] = `
<div>
  <div class="transition" style="display: flex; margin-top: 44px;"><img style="width: 180px; height: 180px; min-width: 180px; min-height: 180px; border-radius: 50%; object-fit: cover;">
    <div style="margin-left: 28px;">
      <h2 style="margin-top: 16px;">
        undefined. firstname lastname&nbsp;<span style="font-weight: 300;">age</span></h2>
      <p class="location" style="font-weight: 400; letter-spacing: -0.09px;">
        streetnumber streetname, city, state
      </p> <button class="email" style="background-color: rgb(226, 227, 236); padding: 9px 14px; border-radius: 999px; align-items: center; display: block;">
        <div style="display: flex; align-items: center;"><i class="far fa-envelope" style="font-size: 16px; margin-right: 8px;"></i> <span style="font-size: 14px; letter-spacing: -0.09px; font-weight: 600;">email</span></div>
      </button> <button style="background-color: rgb(247, 217, 242); font-size: 12px; padding: 9px 15px; border-radius: 999px; font-weight: 700; margin-top: 16px; display: block;">
        JOINED: registrati
      </button>
      <p style="font-size: 14px; font-weight: 500;"><i class="fas fa-phone-alt" style="margin-right: 8px;"></i> <span>cell</span></p>
      <p style="font-size: 14px; font-weight: 500;"><i class="fas fa-mobile-alt" style="margin-right: 8px; margin-left: 4px;"></i> <span>phone</span></p>
    </div>
  </div>
</div>
`;
